numpy/lib/polynomial.py

```diff
@@ -590,14 +606,13 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
     elif cov:
         Vbase = inv(dot(lhs.T, lhs))
         Vbase /= NX.outer(scale, scale)
-        # Some literature ignores the extra -2.0 factor in the denominator, but
-        #  it is included here because the covariance of Multivariate Student-T
-        #  (which is implied by a Bayesian uncertainty analysis) includes it.
-        #  Plus, it gives a slightly more conservative estimate of uncertainty.
-        if len(x) <= order + 2:
-            raise ValueError("the number of data points must exceed order + 2 "
-                             "for Bayesian estimate the covariance matrix")
-        fac = resids / (len(x) - order - 2.0)
+        if absolute_weights:
+            fac = 1.
+        else:
+            if len(x) <= order:
+                raise ValueError("the number of data points must exceed order "
+                                 "for estimate the covariance matrix")
```

Comment:
This error message is not correct any more, it only needs to be true if rescaling is done. Maybe just replace "for estimate" (weird grammar anyway) with "to scale"

Reply:
Still to be done: "for estimate" -> "to scale" (or "in order to be able to scale"